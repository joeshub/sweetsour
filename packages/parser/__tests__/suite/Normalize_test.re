open Jest;

let it = test;

describe("Normalize Test Suite", () => {
  open Expect;

  it("parses normalize.css correctly", () => {
    let cwd = Node.Process.cwd();
    let css = Node.Fs.readFileSync(cwd ++ "/__tests__/suite/normalize.css", `utf8);
    let output = Input.input([| css |], [||])
      |> Lexer.lexer
      |> Parser.parser
      |> Prefixer.prefixer
      |> Printer.printer
      |> Printer.printToStr;

    expect(output) |> toMatchSnapshot
  });
});
