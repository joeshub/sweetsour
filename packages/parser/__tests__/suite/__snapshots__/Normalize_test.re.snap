// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Normalize Test Suite parses normalize.css correctly 1`] = `
"
[RULE_START, STYLE_RULE]
  [SELECTOR, 'html']
  [PROPERTY, 'line-height']
  [VALUE, '1.15']
  [PROPERTY, '-ms-text-size-adjust']
  [VALUE, '100%']
  [PROPERTY, '-webkit-text-size-adjust']
  [VALUE, '100%']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'body']
  [PROPERTY, 'margin']
  [VALUE, '0']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'article']
  [SELECTOR, 'aside']
  [SELECTOR, 'footer']
  [SELECTOR, 'header']
  [SELECTOR, 'nav']
  [SELECTOR, 'section']
  [PROPERTY, 'display']
  [VALUE, 'block']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'h1']
  [PROPERTY, '-webkit-font-size']
  [VALUE, '2em']
  [PROPERTY, '-ms-font-size']
  [VALUE, '2em']
  [PROPERTY, 'font-size']
  [VALUE, '2em']
  [PROPERTY, 'margin']
  [COMPOUND_VALUE_START]
    [VALUE, '0.67em']
    [VALUE, '0']
  [COMPOUND_VALUE_END]
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'figcaption']
  [SELECTOR, 'figure']
  [SELECTOR, 'main']
  [PROPERTY, 'display']
  [VALUE, 'block']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'figure']
  [PROPERTY, 'margin']
  [COMPOUND_VALUE_START]
    [VALUE, '1em']
    [VALUE, '40px']
  [COMPOUND_VALUE_END]
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'hr']
  [PROPERTY, 'box-sizing']
  [VALUE, 'content-box']
  [PROPERTY, 'height']
  [VALUE, '0']
  [PROPERTY, 'overflow']
  [VALUE, 'visible']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'pre']
  [PROPERTY, 'font-family']
  [VALUE, 'monospace']
  [VALUE, 'monospace']
  [PROPERTY, '-webkit-font-size']
  [VALUE, '1em']
  [PROPERTY, '-ms-font-size']
  [VALUE, '1em']
  [PROPERTY, 'font-size']
  [VALUE, '1em']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'a']
  [PROPERTY, 'background-color']
  [VALUE, 'transparent']
  [PROPERTY, '-webkit-text-decoration-skip']
  [VALUE, 'objects']
[RULE_END]
[RULE_START, STYLE_RULE]
  [COMPOUND_SELECTOR_START]
    [SELECTOR, 'abbr']
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'title']
    [ATTRIBUTE_SELECTOR_END]
  [COMPOUND_SELECTOR_END]
  [PROPERTY, 'border-bottom']
  [VALUE, 'none']
  [PROPERTY, 'text-decoration']
  [VALUE, 'underline']
  [PROPERTY, 'text-decoration']
  [COMPOUND_VALUE_START]
    [VALUE, 'underline']
    [VALUE, 'dotted']
  [COMPOUND_VALUE_END]
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'b']
  [SELECTOR, 'strong']
  [PROPERTY, 'font-weight']
  [VALUE, 'inherit']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'b']
  [SELECTOR, 'strong']
  [PROPERTY, 'font-weight']
  [VALUE, 'bolder']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'code']
  [SELECTOR, 'kbd']
  [SELECTOR, 'samp']
  [PROPERTY, 'font-family']
  [VALUE, 'monospace']
  [VALUE, 'monospace']
  [PROPERTY, '-webkit-font-size']
  [VALUE, '1em']
  [PROPERTY, '-ms-font-size']
  [VALUE, '1em']
  [PROPERTY, 'font-size']
  [VALUE, '1em']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'dfn']
  [PROPERTY, 'font-style']
  [VALUE, 'italic']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'mark']
  [PROPERTY, 'background-color']
  [VALUE, '#ff0']
  [PROPERTY, 'color']
  [VALUE, '#000']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'small']
  [PROPERTY, '-webkit-font-size']
  [VALUE, '80%']
  [PROPERTY, '-ms-font-size']
  [VALUE, '80%']
  [PROPERTY, 'font-size']
  [VALUE, '80%']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'sub']
  [SELECTOR, 'sup']
  [PROPERTY, '-webkit-font-size']
  [VALUE, '75%']
  [PROPERTY, '-ms-font-size']
  [VALUE, '75%']
  [PROPERTY, 'font-size']
  [VALUE, '75%']
  [PROPERTY, 'line-height']
  [VALUE, '0']
  [PROPERTY, '-webkit-position']
  [VALUE, 'relative']
  [PROPERTY, 'position']
  [VALUE, 'relative']
  [PROPERTY, 'vertical-align']
  [VALUE, 'baseline']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'sub']
  [PROPERTY, 'bottom']
  [VALUE, '-0.25em']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'sup']
  [PROPERTY, 'top']
  [VALUE, '-0.5em']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'audio']
  [SELECTOR, 'video']
  [PROPERTY, 'display']
  [VALUE, 'inline-block']
[RULE_END]
[RULE_START, STYLE_RULE]
  [COMPOUND_SELECTOR_START]
    [SELECTOR, 'audio']
    [FUNCTION_START, ':not']
      [ATTRIBUTE_SELECTOR_START, 1]
      [ATTRIBUTE_NAME, 'controls']
      [ATTRIBUTE_SELECTOR_END]
    [FUNCTION_END]
  [COMPOUND_SELECTOR_END]
  [PROPERTY, 'display']
  [VALUE, 'none']
  [PROPERTY, 'height']
  [VALUE, '0']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'img']
  [PROPERTY, '-webkit-border-style']
  [VALUE, 'none']
  [PROPERTY, '-ms-border-style']
  [VALUE, 'none']
  [PROPERTY, 'border-style']
  [VALUE, 'none']
[RULE_END]
[RULE_START, STYLE_RULE]
  [COMPOUND_SELECTOR_START]
    [SELECTOR, 'svg']
    [FUNCTION_START, ':not']
      [SELECTOR, ':root']
    [FUNCTION_END]
  [COMPOUND_SELECTOR_END]
  [PROPERTY, 'overflow']
  [VALUE, 'hidden']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'button']
  [SELECTOR, 'input']
  [SELECTOR, 'optgroup']
  [SELECTOR, 'select']
  [SELECTOR, 'textarea']
  [PROPERTY, 'font-family']
  [VALUE, 'sans-serif']
  [PROPERTY, '-webkit-font-size']
  [VALUE, '100%']
  [PROPERTY, '-ms-font-size']
  [VALUE, '100%']
  [PROPERTY, 'font-size']
  [VALUE, '100%']
  [PROPERTY, 'line-height']
  [VALUE, '1.15']
  [PROPERTY, 'margin']
  [VALUE, '0']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'button']
  [SELECTOR, 'input']
  [PROPERTY, 'overflow']
  [VALUE, 'visible']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'button']
  [SELECTOR, 'select']
  [PROPERTY, 'text-transform']
  [VALUE, 'none']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'button']
  [COMPOUND_SELECTOR_START]
    [SELECTOR, 'html']
    [SPACE_COMBINATOR]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"button\\"']
    [ATTRIBUTE_SELECTOR_END]
  [COMPOUND_SELECTOR_END]
  [ATTRIBUTE_SELECTOR_START, 1]
  [ATTRIBUTE_NAME, 'type']
  [ATTRIBUTE_OPERATOR, '=']
  [ATTRIBUTE_VALUE, '\\"reset\\"']
  [ATTRIBUTE_SELECTOR_END]
  [ATTRIBUTE_SELECTOR_START, 1]
  [ATTRIBUTE_NAME, 'type']
  [ATTRIBUTE_OPERATOR, '=']
  [ATTRIBUTE_VALUE, '\\"submit\\"']
  [ATTRIBUTE_SELECTOR_END]
  [PROPERTY, '-webkit-appearance']
  [VALUE, 'button']
[RULE_END]
[RULE_START, STYLE_RULE]
  [COMPOUND_SELECTOR_START]
    [SELECTOR, 'button']
    [SELECTOR, ':-moz-focus-inner']
  [COMPOUND_SELECTOR_END]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"button\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-moz-focus-inner']
  [COMPOUND_SELECTOR_END]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"reset\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-moz-focus-inner']
  [COMPOUND_SELECTOR_END]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"submit\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-moz-focus-inner']
  [COMPOUND_SELECTOR_END]
  [PROPERTY, '-webkit-border-style']
  [VALUE, 'none']
  [PROPERTY, '-ms-border-style']
  [VALUE, 'none']
  [PROPERTY, 'border-style']
  [VALUE, 'none']
  [PROPERTY, '-webkit-padding']
  [VALUE, '0']
  [PROPERTY, 'padding']
  [VALUE, '0']
[RULE_END]
[RULE_START, STYLE_RULE]
  [COMPOUND_SELECTOR_START]
    [SELECTOR, 'button']
    [SELECTOR, ':-moz-focusring']
  [COMPOUND_SELECTOR_END]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"button\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-moz-focusring']
  [COMPOUND_SELECTOR_END]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"reset\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-moz-focusring']
  [COMPOUND_SELECTOR_END]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"submit\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-moz-focusring']
  [COMPOUND_SELECTOR_END]
  [PROPERTY, 'outline']
  [COMPOUND_VALUE_START]
    [VALUE, '1px']
    [VALUE, 'dotted']
    [VALUE, 'ButtonText']
  [COMPOUND_VALUE_END]
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'fieldset']
  [PROPERTY, '-webkit-padding']
  [COMPOUND_VALUE_START]
    [VALUE, '0.35em']
    [VALUE, '0.75em']
    [VALUE, '0.625em']
  [COMPOUND_VALUE_END]
  [PROPERTY, 'padding']
  [COMPOUND_VALUE_START]
    [VALUE, '0.35em']
    [VALUE, '0.75em']
    [VALUE, '0.625em']
  [COMPOUND_VALUE_END]
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'legend']
  [PROPERTY, 'box-sizing']
  [VALUE, 'border-box']
  [PROPERTY, 'color']
  [VALUE, 'inherit']
  [PROPERTY, 'display']
  [VALUE, 'table']
  [PROPERTY, 'max-width']
  [VALUE, '100%']
  [PROPERTY, '-webkit-padding']
  [VALUE, '0']
  [PROPERTY, 'padding']
  [VALUE, '0']
  [PROPERTY, '-webkit-white-space']
  [VALUE, 'normal']
  [PROPERTY, 'white-space']
  [VALUE, 'normal']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'progress']
  [PROPERTY, 'display']
  [VALUE, 'inline-block']
  [PROPERTY, 'vertical-align']
  [VALUE, 'baseline']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'textarea']
  [PROPERTY, 'overflow']
  [VALUE, 'auto']
[RULE_END]
[RULE_START, STYLE_RULE]
  [ATTRIBUTE_SELECTOR_START, 1]
  [ATTRIBUTE_NAME, 'type']
  [ATTRIBUTE_OPERATOR, '=']
  [ATTRIBUTE_VALUE, '\\"checkbox\\"']
  [ATTRIBUTE_SELECTOR_END]
  [ATTRIBUTE_SELECTOR_START, 1]
  [ATTRIBUTE_NAME, 'type']
  [ATTRIBUTE_OPERATOR, '=']
  [ATTRIBUTE_VALUE, '\\"radio\\"']
  [ATTRIBUTE_SELECTOR_END]
  [PROPERTY, 'box-sizing']
  [VALUE, 'border-box']
  [PROPERTY, '-webkit-padding']
  [VALUE, '0']
  [PROPERTY, 'padding']
  [VALUE, '0']
[RULE_END]
[RULE_START, STYLE_RULE]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"number\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-webkit-inner-spin-button']
  [COMPOUND_SELECTOR_END]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"number\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-webkit-outer-spin-button']
  [COMPOUND_SELECTOR_END]
  [PROPERTY, 'height']
  [VALUE, 'auto']
[RULE_END]
[RULE_START, STYLE_RULE]
  [ATTRIBUTE_SELECTOR_START, 1]
  [ATTRIBUTE_NAME, 'type']
  [ATTRIBUTE_OPERATOR, '=']
  [ATTRIBUTE_VALUE, '\\"search\\"']
  [ATTRIBUTE_SELECTOR_END]
  [PROPERTY, '-webkit-appearance']
  [VALUE, 'textfield']
  [PROPERTY, 'outline-offset']
  [VALUE, '-2px']
[RULE_END]
[RULE_START, STYLE_RULE]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"search\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-webkit-search-cancel-button']
  [COMPOUND_SELECTOR_END]
  [COMPOUND_SELECTOR_START]
    [ATTRIBUTE_SELECTOR_START, 1]
    [ATTRIBUTE_NAME, 'type']
    [ATTRIBUTE_OPERATOR, '=']
    [ATTRIBUTE_VALUE, '\\"search\\"']
    [ATTRIBUTE_SELECTOR_END]
    [SELECTOR, ':-webkit-search-decoration']
  [COMPOUND_SELECTOR_END]
  [PROPERTY, '-webkit-appearance']
  [VALUE, 'none']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, ':-webkit-file-upload-button']
  [PROPERTY, '-webkit-appearance']
  [VALUE, 'button']
  [PROPERTY, 'font']
  [VALUE, 'inherit']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'details']
  [SELECTOR, 'menu']
  [PROPERTY, 'display']
  [VALUE, 'block']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'summary']
  [PROPERTY, 'display']
  [VALUE, 'list-item']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'canvas']
  [PROPERTY, 'display']
  [VALUE, 'inline-block']
[RULE_END]
[RULE_START, STYLE_RULE]
  [SELECTOR, 'template']
  [PROPERTY, 'display']
  [VALUE, 'none']
[RULE_END]
[RULE_START, STYLE_RULE]
  [ATTRIBUTE_SELECTOR_START, 1]
  [ATTRIBUTE_NAME, 'hidden']
  [ATTRIBUTE_SELECTOR_END]
  [PROPERTY, 'display']
  [VALUE, 'none']
[RULE_END]"
`;
